TESTS/mbedmicro-rtos-mbed/timer/main.cpp

```diff
@@ -0,0 +1,872 @@
+
+static Timer *p_timer;
+
+/* Global variable used to simulate passage of time
+ * in case when timer which uses user ticker is tested.
+ */
+static uint32_t curr_ticker_us_val;
+
+/* User ticker interface function. */
+static void my_interface_init()
+{
+    /* do nothing. */
+}
+
+/* User ticker interface function - only this
+ * ticker interface function is used by Timer API. */
+static uint32_t my_ticker_read(void)
+{
+    /* Simulate elapsed time. */
+    return curr_ticker_us_val;
+}
+
+/* User ticker interface function. */
+static void my_disable_interrupt(void)
+{
+    /* do nothing. */
+}
+
+/* User ticker interface function. */
+static void my_clear_interrupt(void)
+{
+    /* do nothing. */
+}
+
+/* User ticker interface function. */
+static void my_set_interrupt(timestamp_t timestamp)
+{
+    /* do nothing. */
+}
+
+/* User ticker interface function. */
+static void my_fire_interrupt(void)
+{
+    /* do nothing. */
+}
+
+/* User ticker event queue. */
+static ticker_event_queue_t my_events = { 0 };
+
+/* User ticker interface data. */
+static const ticker_interface_t us_interface = {
+    .init = my_interface_init,
+    .read = my_ticker_read, /* Only this function is used by the Timer. */
+    .disable_interrupt = my_disable_interrupt,
+    .clear_interrupt = my_clear_interrupt,
+    .set_interrupt = my_set_interrupt,
+    .fire_interrupt = my_fire_interrupt,
+};
+
+/* User ticker data structure. */
+static const ticker_data_t us_data = {
+    .interface = &us_interface,
+    .queue = &my_events
+};
+
+/* Function which returns user ticker data. */
+const ticker_data_t* get_user_ticker_data(void)
+{
+    return &us_data;
+}
+
+/* Initialisation of the Timer object which uses
+ * ticker data provided by the user.
+ *
+ * */
+utest::v1::status_t timer_user_ticker_setup_h(const Case *const source, const size_t index_of_case)
+{
+    p_timer = new Timer(get_user_ticker_data());
+
+    /* Check if Timer object has been created. */
+    TEST_ASSERT_NOT_NULL(p_timer);
+
+    return greentea_case_setup_handler(source, index_of_case);
+}
+
+/* Initialisation of the Timer object which uses
+ * default os ticker data.
+ *
+ * */
+utest::v1::status_t timer_os_ticker_setup_h(const Case *const source, const size_t index_of_case)
+{
+    p_timer = new Timer();
+
+    /* Check if Timer object has been created. */
+    TEST_ASSERT_NOT_NULL(p_timer);
+
+    return greentea_case_setup_handler(source, index_of_case);
+}
+
+/* Test finalisation.
+ *
+ * */
+utest::v1::status_t cleanup_h(const Case *const source, const size_t passed, const size_t failed, const failure_t reason)
+{
+    delete p_timer;
+
+    return greentea_case_teardown_handler(source, passed, failed, reason);
+}
+
+/* Test which verify that timer is stopped after
+ * creation.
+ *
+ * Given Timer has been successfully created.
+ * When read of timer elapsed time is requested.
+ * Then result is always 0.
+ */
+void test_timer_creation()
+{
+    float read_val;
```

Comment:
Fixed - removed local variables to hold measured time. Instead call read function inside assertion.